src/app/services/graph.service.js

```diff
@@ -191,6 +192,15 @@ angular
                         'background-color': '#ff5688',
                     }
                 },
+                // apply custom colors on top of all defaults, but allow the selected and hidden options below to override this
+                // TODO: How to get the custom color from the dictionary mapped below
+                {
+                    selector: 'node[node_color_indicator=1]', //this is correctly mapped to the node_color attribute
+                    style: {
+                        'background-color': '#000000', //placeholder hex color, this needs to be dynamic
```

Comment:
```suggestion
                        'background-color': node_color_variable, //placeholder hex color, this needs to be dynamic
```

Reply:
Even if this works to create a dynamic color mapping, it can only do it with ONE color at a time, not many concurrently. 

I suspect the selector for nodes with the custom color config will require creating a dynamic mapping of unique nodes to unique colors. Pseudo-code below.
```javascript
                {
                    selector: 'node[name="dim_parts"]',
                    style: {
                        'background-color': node_color_var[name="dim_parts"], // dynamic parameter
                    }
                },

```